power_grid_model_c_example/main.c

```diff
@@ -66,39 +67,39 @@ int main(int argc, char** argv) {
     // Using the buffer helper function is more convenient but with some overhead.
 
     // node attribute, we use pointer cast
-    size_t node_id_offset = PGM_meta_attribute_offset(handle, "input", "node", "id");
-    size_t node_u_rated_offset = PGM_meta_attribute_offset(handle, "input", "node", "u_rated");
+    size_t node_id_offset = PGM_meta_attribute_offset(handle, PGM_zdef_input_node_id);
+    size_t node_u_rated_offset = PGM_meta_attribute_offset(handle, PGM_zdef_input_node_u_rated);
     // pointer cast of offset
     *(PGM_ID*)((char*)node_input + node_id_offset) = 1;
     *(double*)((char*)node_input + node_u_rated_offset) = 10e3;  // 10 kV node
 
     // source attribute, we use helper function
     // set to NaN for all values, it is recommended for input and update buffers
-    PGM_buffer_set_nan(handle, "input", "source", source_input, 1);
+    PGM_buffer_set_nan(handle, PGM_zdef_input_source, source_input, 0, 1);
```

Comment:
I am in favor of having it so that the user does need to do `char` pointer arithmetic. Inside the core we can cast the pointer to actual `struct` class and do direct `ptr + offset`. Otherwise the user has to do `(void*)((char*)ptr + offset * size)`.